src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,204 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class TaskList {
+
+    private ArrayList<Task> lstOfItems;
+
+    public TaskList(File file) {
+        this.lstOfItems = new ArrayList<>();
+        try {
+            Scanner fileScanner = new Scanner(file);
+            int noOfTasks = Integer.parseInt(fileScanner.nextLine());
+            for (int i = 0; i < noOfTasks; i++) {
+                String curr = fileScanner.nextLine();
+                parseTask(curr);
+            }
+        } catch (FileNotFoundException err) {
+            System.out.println(err);
+        }
+    }
+
+    public TaskList() {
+        this.lstOfItems = new ArrayList<>();
+    }
+
+    /**
+     * Parses the previously saved tasks that were keyed in, in the previous iteration.
+     * @param currentTask the saved task in the File in the filePath.
+     */
+    public void parseTask(String currentTask) {
+        Task task = null;
+        if (currentTask.charAt(1) == 'T') {
+            task = new Todo(currentTask.substring(7));
+        } else if (currentTask.charAt(1) == 'D') {
+            String[] split = currentTask.split("by: ");
```

Comment:
You may consider renaming `split` into a plural form since it is a String array.